scheduler/scheduler.go

```diff
@@ -583,11 +584,14 @@ func (s *scheduler) gatherMetricsAndPlugins(wf *schedulerWorkflow) ([]core.Metri
 	)
 
 	for _, m := range wf.metrics {
-		mts = append(mts, &metric{
-			namespace: m.Namespace(),
-			version:   m.Version(),
-			config:    wf.configTree.Get(m.Namespace()),
-		})
+		nss := s.metricManager.ExpandWildCards(m.Namespace())
```

Comment:
ExpandWildCards is calling control.GetMatchedMetrics and ignoring the error.  Consider returning the err so we can use the metrics directly from the workflow if err != nil.  

I found this issue when I was allowed to create a task with a single invalid metric.  The task creation should have failed with an error indicating an invalid metric was provided.